Mention NetFx40_LegacySecurityPolicy for PublicSign

KirillOsenkov opened the pull request:
Mention that PublicSign doesn't work if the app.config specifies `<NetFx40_LegacySecurityPolicy enabled="true" />`

KirillOsenkov committed:
Update public-signing.md

KirillOsenkov commented:
FYI @jeffhandley @inthemedium ask me how I learned about this one

JustinSchneiderPBI commented on docs/project/public-signing.md:
```diff
@@ -17,6 +17,7 @@ Known issues when debugging and testing public signed assemblies on .NET Framewo
 - You will not be able to load the assembly in an AppDomain where shadow copying is turned on.
 - You will not be able to load the assembly in a partially trusted AppDomain
 - You will not be able to pre-compile ASP.NET applications
+- You will not be able to load the assembly if the `app.config` file specifies `<configuration`/`<runtime>`/`<NetFx40_LegacySecurityPolicy enabled="true" />`
```
"may" not? Remember there's some context here that we haven't identified why some environments are fine with this.

KirillOsenkov committed:
Specify "may" because we've seen it work on some machines but not others.

It's not clear to use what the difference is other than a hunch that it fails on Windows Server but works fine on Windows Client OS.

KirillOsenkov committed:
Add missing angle bracket
